pkg/tsdb/cloudwatch/metric_find_query.go

```diff
@@ -111,6 +112,9 @@ func (e *cloudWatchExecutor) handleGetEc2InstanceAttribute(ctx context.Context,
 				v := reflect.ValueOf(instance)
 				for _, key := range attributePath {
 					if v.Kind() == reflect.Ptr {
+						if v.IsNil() {
+							continue instanceLoop
+						}
```

Comment:
Purely optional, but _if_ you wanted to avoid the outer loop (they always feel a bit dirty to me despite being perfectly legitimate) you could extract the contents of this `else` block into a function looking something like

```go
func getInstanceAttributeValue(attributeName string, instance *ec2.Instance) (value string, found bool, err error) {
...
}
```

But like I said, not necessary by any means.